### PR TITLE
fix must material for Contact "C"

### PR DESCRIPTION
**Pull request by chronogenexx**
need https://github.com/Fluorohydride/ygopro-core/pull/412.

### CODE REVIEW

File: utility.lua [hunk 1]
```diff
@@ -2042,7 +2020,7 @@ function Auxiliary.LinkTarget(f,minc,maxc,gf)
 					if not Auxiliary.LConditionFilter(lmat,f,c,e) then return false end
 					mg:AddCard(lmat)
 				end
-				local fg=Auxiliary.GetMustMaterialGroup(tp,EFFECT_MUST_BE_LMATERIAL)
+				local fg=Duel.GetMustMaterial(tp,EFFECT_MUST_BE_XMATERIAL)
```
Comment (mercury233) [inline utility.lua:2023]: LMATERIAL -> XMATERIAL ?

**Comment by mercury233** [comment]
I prefer to keep the aux function for compatibility

File: utility.lua (hunk 1 above)
Comment (chronogenexx) [reply utility.lua]: https://github.com/Fluorohydride/ygopro-scripts/pull/1727/commits/450a34390667068dc58ab4ab1a849d4d0282c47b fixed.

**Comment by chronogenexx** [comment]
https://github.com/Fluorohydride/ygopro-scripts/pull/1727/commits/73bf8c8bb9a82375a34472701faee7221409faef is this ok?

**Comment by purerosefallen** [comment]
plz add `Auxiliary.GetMustMaterialGroup` back.

**Comment by chronogenexx** [comment]
`Auxiliary.GetMustMaterialGroup` is used utility.lua only.
so, there is no problem to instead `Duel.GetMustMaterial`.

**Comment by mercury233** [comment]
> `Auxiliary.GetMustMaterialGroup` is used utility.lua only. so, there is no problem to instead `Duel.GetMustMaterial`.

there may be some custom scripts using that directly, ~~but I prefer to let them update their scripts~~

**Comment by salix5** [comment]
[contact.zip](https://github.com/Fluorohydride/ygopro-scripts/files/10491475/contact.zip)

contact_bug.yrp
current version
BUG: The player can Syncho Summon Level 5 monster.

contact_correct.yrp
correct version
The player cannot Synchro Summon.
contact2


Since one year has passed and the bug is still unsolved, I think we can merge this one as a workaround.

**Pull request merged**